From 4849048b6782d4eff46aa134e2cb010ffbb6d43e Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Fri, 13 Dec 2024 12:21:35 +0100
Subject: [PATCH] arm64: dts: st: usb2phy: adopt once cell clock bindings on
 stm32mp211

Distinguish OHCI and PLL clocks on stm32mp211, by using one cell clock
bindings. The OHCI clock must be enabled specifically during resume from
low power.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: I4732a9575fe1f778ee116ed666abb8d992501295
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/424748
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christian BRUEL <christian.bruel@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp211.dtsi b/arch/arm64/boot/dts/st/stm32mp211.dtsi
index 6911ce15e143..ef95ba2264b7 100644
--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -255,7 +255,7 @@ timer {
 	usb2_phy1: usb2-phy1 {
 		compatible      = "st,stm32mp21-usb2phy";
 		#phy-cells      = <0>;
-		#clock-cells    = <0>;
+		#clock-cells    = <1>;
 		st,syscfg       = <&syscfg 0x2400>;
 		clocks          = <&rcc CK_KER_USB2PHY1>;
 		resets          = <&rcc USB2PHY1_R>;
@@ -265,7 +265,7 @@ usb2_phy1: usb2-phy1 {
 	usb2_phy2: usb2-phy2 {
 		compatible      = "st,stm32mp21-usb2phy";
 		#phy-cells      = <0>;
-		#clock-cells    = <0>;
+		#clock-cells    = <1>;
 		st,syscfg       = <&syscfg 0x2800>;
 		clocks          = <&rcc CK_KER_USB2PHY2EN>;
 		resets          = <&rcc USB2PHY2_R>;
@@ -1880,7 +1880,7 @@ usbh: usb@482e0000 {
 				usbh_ohci: usb@482e0000 {
 					compatible = "generic-ohci";
 					reg = <0x482e0000 0x1000>;
-					clocks = <&usb2_phy1>, <&rcc CK_BUS_USBHOHCI>;
+					clocks = <&usb2_phy1 1>, <&rcc CK_BUS_USBHOHCI>;
 					resets = <&rcc USBH_R>;
 					interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
 					phys = <&usb2_phy1>;
@@ -1891,7 +1891,7 @@ usbh_ohci: usb@482e0000 {
 				usbh_ehci: usb@482f0000 {
 					compatible = "generic-ehci";
 					reg = <0x482f0000 0x1000>;
-					clocks = <&usb2_phy1>, <&rcc CK_BUS_USBHEHCI>;
+					clocks = <&usb2_phy1 0>, <&rcc CK_BUS_USBHEHCI>;
 					resets = <&rcc USBH_R>;
 					interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
 					companion = <&usbh_ohci>;
@@ -1904,7 +1904,7 @@ usbh_ehci: usb@482f0000 {
 			usbotg_hs: usb@48300000 {
 				compatible = "st,stm32mp21-hsotg", "snps,dwc2";
 				reg = <0x48300000 0x0 0x10000>;
-				clocks = <&rcc CK_BUS_OTG>, <&usb2_phy2>;
+				clocks = <&rcc CK_BUS_OTG>, <&usb2_phy2 0>;
 				clock-names = "otg", "utmi";
 				resets = <&rcc OTG_R>;
 				reset-names = "dwc2";
-- 
2.53.0

